client/blocks/comment-detail/index.jsx

```diff
@@ -80,18 +81,40 @@ export class CommentDetail extends Component {
 
 	edit = () => noop;
 
+	getCommentObject = () => ( {
+		commentId: this.props.commentId,
+		isLiked: this.props.commentIsLiked,
+		postId: this.props.postId,
+		status: this.props.commentStatus,
+	} );
+
 	toggleApprove = () => {
-		const { commentId, commentStatus, postId, setCommentStatus } = this.props;
-		setCommentStatus( commentId, postId, 'approved' === commentStatus ? 'unapproved' : 'approved' );
+		const { commentStatus, setCommentStatus } = this.props;
+		const shouldPersist = 'approved' === commentStatus || 'unapproved' === commentStatus;
+
+		setCommentStatus( this.getCommentObject(), 'approved' === commentStatus ? 'unapproved' : 'approved', {
```

Comment:
just a quick note that this can seem ambiguous here. ternary precedence varies by language so we could anticipate this code doing one of two things:

  - either return `unapproved` or `approved` based on whether `commentStatus` === approved
  - return a boolean if `approved` === (either `unapproved` or `approved` based on whether `commentStatus` is truthy)

a pair of parens would remove all ambiguity